src/interface-validator.ts

```diff
@@ -58,6 +58,15 @@ export function getMismatchedFields(
       return (isOptional) ? [] : [valueKey];
     }
 
+    if (_.isArray(fieldSpecs[specName])) {
+      return values[valueKey].map((arrayValue, index) => {
+        return getMismatchedFields(
```

Comment:
This part can be modified to support primitive arrays. I imagine the way to specify primitive arrays to be like `['number']` or `['boolean']`, following your test example. Then to add primitive support here, you need to:

1. Extract the primitive check from Line 76 to Line 111 to a function accepting the primitive type and a value.
2. Check here if the `fieldSpecs[specName][0]` is a string, indicating a primitive type. If it is, use the extracted function to check. Otherwise, call `getMismatchedFields`.

Reply:
Also, in the extracted function, check if the primitive type is one of the expected types, i.e. the string name is correct. You can throw a TypeError if it isn't. This is to provide runtime safety.